src/liteflow/executor.py

```diff
@@ -26,17 +26,184 @@ def submit(self, fn: Callable, *args, **kwargs) -> Future:
+            from liteflow.flow import TaskOutput, NextTask
+            from liteflow.context import Context
+            import logging
+            import inspect
+            
+            # Create a context with the provided data
+            ctx = Context()
+            if context_data:
+                for key, value in context_data.items():
+                    ctx.set(key, value)
+            
+            # Create a logger
+            logger = logging.getLogger(logger_name or "liteflow.ray")
+            
+            # Execute the action
+            try:
+                # Check if function accepts inputs parameter
+                sig = inspect.signature(action_fn)
+                params = list(sig.parameters.keys())
+                
+                if len(params) > 1 and params[1] == "inputs":
+                    result = action_fn(ctx, inputs=task_inputs)
+                else:
+                    result = action_fn(ctx)
+                
+                # Convert result to dict if it's a TaskOutput
+                if isinstance(result, TaskOutput):
+                    next_tasks = None
+                    if result.next_tasks:
+                        next_tasks = [
+                            {
+                                'id': nt.id,
+                                'inputs': nt.inputs,
+                                'spawn_another': nt.spawn_another
+                            }
+                            for nt in result.next_tasks
+                        ]
+                    
+                    return {
+                        'output': result.output,
+                        'next_tasks': next_tasks,
+                        'is_task_output': True
+                    }
+                
+                return result
+            except Exception as e:
+                import traceback
+                error_info = {
+                    'error': str(e),
+                    'traceback': traceback.format_exc(),
+                    'is_error': True
+                }
+                return error_info
+        
+        self._ray_execute_task = ray_execute_task
+            
+    def submit(self, fn: Callable, *args, **kwargs) -> Future:
+        """Submit a task to Ray for execution and return a Future-like object"""
+        # Extract arguments
+        action = args[0]  # The task function
+        next_task = args[1]  # NextTask object
+        context = args[2]  # Context object
+        logger = args[3] if len(args) > 3 else None  # Logger
+        
+        # Extract context data - only include serializable data
+        context_data = {}
+        for key in context.states:
+            try:
+                value = context.get(key, None)
+                if value is not None:
+                    context_data[key] = value
+            except:
+                pass
+        
+        # Get logger name
+        logger_name = logger.name if logger else None
+        
```

Comment:
RayExecutor only needs to implement the same execution function as PoolExecutor, and there is no need to introduce the context of flow. The transfer of context has been implemented in the flow framework.

Reply:
> RayExecutor only needs to implement the same execution function as PoolExecutor, and there is no need to introduce the context of flow. The transfer of context has been implemented in the flow framework.

[agentfarmx](https://github.com/apps/agentfarmx)